lib/storage/index.ts

```diff
@@ -1,13 +1,52 @@
+import * as Logger from '../Logger';
+
 import PlatformStorage from './platforms';
 import InstanceSync from './InstanceSync';
+import NoopProvider from './providers/NoopProvider';
 import type StorageProvider from './providers/types';
 
-const provider = PlatformStorage;
+let provider = PlatformStorage;
 let shouldKeepInstancesSync = false;
+let finishInitalization: (value?: unknown) => void;
+const initPromise = new Promise((resolve) => {
+    finishInitalization = resolve;
+});
 
 type Storage = {
     getStorageProvider: () => StorageProvider;
-} & StorageProvider;
+} & Omit<StorageProvider, 'name'>;
```

Comment:
Why are we omititng the name here?

Reply:
@danieldoglas `StorageProvider` contains `name` property because this property is used in logger (to know which provider threw the error).

And `Storage` acts as a gate, i. e. underhood this class/object can substitute internal providers, but the gateway will be always the same. So I thought it senseless to have this property on a gateway level 🤷‍♂️